pkg/operator/ceph/object/admin.go

```diff
@@ -110,8 +111,57 @@ func runAdminCommand(c *Context, expectJSON bool, args ...string) (string, error
 			fmt.Sprintf("--rgw-zone=%s", c.Zone),
 		}
 
-		return RunAdminCommandNoMultisite(c, expectJSON, append(args, options...)...)
+		args = append(args, options...)
 	}
 
-	return RunAdminCommandNoMultisite(c, expectJSON, args...)
+	// work around https://github.com/rook/rook/issues/7573 (FIFO file I/O issue)
+	result, err := RunAdminCommandNoMultisite(c, expectJSON, args...)
+	if isFifoFileIOError(err) {
```

Comment:
nit: IMO the more common pattern is to check for err != nil before evaluating an error, rather than checking for err == nil inside the helper.
```suggestion
	if err != nil && isFifoFileIOError(err) {
```